src/move/lp_coin_manager/sources/lp_coin_manager.move

```diff
@@ -0,0 +1,57 @@
+module lp_coin_manager::lp_coin_manager {
+    use aptos_framework::coin::{Self, Coin, BurnCapability, FreezeCapability, MintCapability};
+    use std::signer::{address_of};
+
+    #[resource_group = ObjectGroup]
+    struct LPCoinCapabilities<phantom CoinType, phantom LPCoinType> has key {
```

Comment:
```suggestion
    struct LPCoinCapabilities<phantom LPCoinType> has key {
```

Reply:
So I did this specifically to ensure that a malicious user doesn't ever send in incorrect generics to a function that requires either a `CoinType` or a `LPCoinType`. It enforces us and the user to be consistent because there won't ever exist a type `E, E_LP` where the Coins don't "match".

For example, right now the user can call `swap<B, Q, LP>` with incorrect generics that don't actually match and then the `mint_lp_coins` would mint the wrong LP coin.

If/when we eventually have some `provide_liquidity<E, E_LP>` function, if a user provides an incorrect `E_LP` type they could potentially get LP_coins for a different, non-matching coin

Does that make sense? I could be wrong here but just figured it was worth being safe and forcing the types to be paired.

Reply:
On the topic of type checking I took some notes in the testing tasks that we might want to make use of typeinfo

Can follow up with that later since is now flagged, but glad we're thinking about this early on